cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/ExecutorStateProcessor.java

```diff
@@ -0,0 +1,115 @@
+        }
+        return executorState.get(TOTAL_DATA_TO_MOVE_KEY).asInt();
+    }
+
+    /**
+     * Retrieves the amount of data that has already been moved from the provided `executorState` JSON object.
+     * The value is obtained from the "finishedDataMovement" field.
+     *
+     * @param executorState The `JsonNode` object containing the state of the executor,
+     *                      from which the finished data movement is extracted.
+     * @return The amount of data that has already been moved, in megabytes.
+     */
+    public static Integer getFinishedDataMovement(JsonNode executorState) {
+        if (!executorState.has(FINISHED_DATA_MOVEMENT_KEY)) {
+            throw new IllegalArgumentException(String.format("Executor State does not contain required '%s' field.", FINISHED_DATA_MOVEMENT_KEY));
+        }
+        return executorState.get(FINISHED_DATA_MOVEMENT_KEY).asInt();
+    }
+
+    /**
+     * Extracts the task start time from the provided `executorState` JSON object.
+     * The task start time is extracted from the "triggeredTaskReason" field, which contains a
+     * timestamp in ISO 8601 format. The timestamp is then parsed into seconds since the Unix epoch.
+     *
+     * @param executorState The `JsonNode` object containing the state of the executor,
+     *                      from which the task start time will be extracted.
+     * @return The task start time in seconds since the Unix epoch.
+     */
+    public static Integer getTaskStartTime(JsonNode executorState) {
+        if (!executorState.has(TRIGGERED_TASK_REASON_KEY)) {
+            throw new IllegalArgumentException(String.format("Executor State does not contain required '%s' field.", TRIGGERED_TASK_REASON_KEY));
+        }
+        String triggeredTaskReason = executorState.get(TRIGGERED_TASK_REASON_KEY).asText();
+        // Extract the timestamp from the string, assuming it's in ISO 8601 format
+        String dateString = extractDateFromTriggeredTaskReason(triggeredTaskReason);
+        return (int) parseDateToSeconds(dateString);
+    }
+
+    /**
+     * Method to parse date-time string in ISO 8601 into time in seconds since Unix epoch.
+     *
+     * @param dateString Date-time string in ISO 8601 format.
+     *
+     * @return Seconds since Unix epoch.
+     */
+    private static long parseDateToSeconds(String dateString) {
+        // Validate the date format
+        if (dateString == null || dateString.isEmpty()) {
+            throw new IllegalArgumentException("Invalid date string.");
+        }
+        // Use Instant to parse the date-time string in ISO 8601 format
+        Instant instant = Instant.parse(dateString);
+        return instant.getEpochSecond();  // Convert to seconds
+    }
+
+    /**
+     * Extracts the ISO 8601 date-time string from a Cruise Control task's reason string.
+     * The date-time is expected to be in the format "%s (Client: %s, Date: %s)", where
```

Comment:
It's the reason to be in this format, not the date-time. I mean the sentence should be "The reason is expected to be ..., where the date ..."

Reply:
Fixed.